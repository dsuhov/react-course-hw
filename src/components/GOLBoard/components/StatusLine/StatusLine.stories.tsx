import React from "react";
import { StatusLine } from "./StatusLine";

import { Provider } from "react-redux";
import { store } from "@/rdx/store";

export default {
  title: "Status Line",
};

export const statusLine = () => {
  return (
    <Provider store={store}>
      <StatusLine />
    </Provider>
  );
};
