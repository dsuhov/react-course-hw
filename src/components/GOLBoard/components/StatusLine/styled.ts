import styled from "@emotion/styled";

export const StatusLineWrapper = styled.div`
  padding-top: 10px;
  display: flex;
  justify-content: center;
  font-size: 0.9em;
  color: #505050;
`;

export const StatusSize = styled.div`
  margin-right: 20px;
`;

export const GenerationStatus = styled.div`
  margin-left: 10px;
`;
