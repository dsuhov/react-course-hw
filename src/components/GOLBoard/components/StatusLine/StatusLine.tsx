import React from "react";
import { StatusLineWrapper, StatusSize, GenerationStatus } from "./styled";

interface StatusLineProps {
  size: [number, number];
  interval: number;
}

function areEqual(prevProps: StatusLineProps, nextProps: StatusLineProps) {
  const isEqInterval = prevProps.interval === nextProps.interval;
  const isSizeEqual =
    prevProps.size[0] === nextProps.size[0] &&
    prevProps.size[1] === nextProps.size[1];

  return isEqInterval && isSizeEqual;
}

export const StatusLine = React.memo<StatusLineProps>(({ size, interval }) => {
  return (
    <StatusLineWrapper>
      <StatusSize>
        Size: {size[0]} x {size[1]}
      </StatusSize>
      <div>
        Tick Interval: <b>{interval}</b> ms
      </div>
      <GenerationStatus>
        Generation: <b>5</b>
      </GenerationStatus>
    </StatusLineWrapper>
  );
}, areEqual);

StatusLine.displayName = "StatusLine";
