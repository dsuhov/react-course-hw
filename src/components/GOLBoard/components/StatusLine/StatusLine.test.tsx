import React from "react";
import renderer from "react-test-renderer";
import { StatusLine } from "./StatusLine";

describe("<StatusLine />  tests", () => {
  const defaultParams: {
    size: [number, number];
    interval: number;
  } = {
    size: [25, 31],
    interval: 548,
  };

  it("<StatusLine /> snapshot", () => {
    expect(
      renderer.create(<StatusLine {...defaultParams} />).toJSON()
    ).toMatchSnapshot();
  });
});
