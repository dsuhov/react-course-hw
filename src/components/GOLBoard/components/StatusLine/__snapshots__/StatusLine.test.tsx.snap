// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusLine />  tests <StatusLine /> snapshot 1`] = `
.emotion-1 {
  padding-top: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 0.9em;
  color: #505050;
}

.emotion-0 {
  margin-right: 20px;
}

<div
  className="emotion-1"
>
  <div
    className="emotion-0"
  >
    Size: 
    25
     x 
    31
  </div>
  <div>
    Tick Interval: 
    <b>
      548
    </b>
     ms
  </div>
</div>
`;
